scalanet/src/io/iohk/scalanet/peergroup/ReqResponseProtocol.scala

```diff
@@ -83,14 +82,14 @@ class ReqResponseProtocol[A, M](
       // sending and subsription are done in parallel to not miss response message by chance
       // alse in case of send failure, any resource will be cleaned
       result <- Task
-        .parMap2(c.sendMessage(messageToSend), subscribeForResponse(c.in, messageToSend.id))(
+        .parMap2(c.sendMessage(messageToSend), subscribeForResponse(c.in.share, messageToSend.id))(
```

Comment:
What's the need for `.share` here? If I understand it correctly, `c.in` will return a new `Observable` that polls the underlying message queue, and `.share` further creates an `Observable` on top of that one which multi-casts among its consumers, however there's only one consumer in `subscribeForResponse` so that's moot.

Reply:
yep you are right, i just wanted here to leave types/scenario as it was without much changes. If we decide that it is worth progressing with that change, then it should be just Observable.